src/test/java/com/mongodb/hibernate/jdbc/MongoStatementTests.java

```diff
@@ -56,24 +65,140 @@ class MongoStatementTests {
+                }""";
+        private final String exampleUpdateMql =
+                """
+                {
+                  update: "members",
+                  updates: [
+                    {
+                      q: {},
+                      u: { $inc: { points: 1 } },
+                      multi: true
+                    }
+                  ]
+                }""";
+
+        @Mock
+        MongoCollection<BsonDocument> mongoCollection;
+
+        @Mock
+        AggregateIterable<BsonDocument> aggregateIterable;
+
+        @Mock
+        MongoCursor<BsonDocument> mongoCursor;
+
+        private ResultSet lastOpenResultSet;
+
+        @BeforeEach
+        void beforeEach() throws SQLException {
+            doReturn(mongoCollection).when(mongoDatabase).getCollection(anyString(), eq(BsonDocument.class));
+            doReturn(aggregateIterable).when(mongoCollection).aggregate(same(clientSession), anyList());
+            doReturn(mongoCursor).when(aggregateIterable).cursor();
+
+            lastOpenResultSet = mongoStatement.executeQuery(exampleQueryMql);
+            assertFalse(lastOpenResultSet.isClosed());
+        }
+
+        @Test
+        void testExecuteQuery() throws SQLException {
+            mongoStatement.executeQuery(exampleQueryMql);
+            assertTrue(lastOpenResultSet.isClosed());
+        }
+
+        @Test
+        void testExecuteUpdate() throws SQLException {
+            doReturn(Document.parse("{n: 10}"))
+                    .when(mongoDatabase)
+                    .runCommand(eq(clientSession), any(BsonDocument.class));
+            mongoStatement.executeUpdate(exampleUpdateMql);
+            assertTrue(lastOpenResultSet.isClosed());
+        }
+
+        @Test
+        void testExecute() throws SQLException {
+            assertThrows(FeatureNotSupportedException.class, () -> mongoStatement.execute(exampleUpdateMql));
+            assertTrue(lastOpenResultSet.isClosed());
+        }
+
+        @Test
+        void testExecuteBatch() throws SQLException {
+            assertThrows(FeatureNotSupportedException.class, () -> mongoStatement.executeBatch());
+            assertTrue(lastOpenResultSet.isClosed());
+        }
+    }
+
+    @Test
+    void testGetProjectStageFieldNames() {
+        var map = Map.of(
+                BsonDocument.parse("{title: 1, publishYear: 1}"), List.of("title", "publishYear", "_id"),
+                BsonDocument.parse("{title: 1, publishYear: 0}"), List.of("title", "_id"),
+                BsonDocument.parse("{title: 1, publishYear: false}"), List.of("title", "_id"),
+                BsonDocument.parse("{title: 1, _id: 0}"), List.of("title"),
+                BsonDocument.parse("{title: 1, _id: false}"), List.of("title"),
+                BsonDocument.parse("{_id: 1, title: 1}"), List.of("_id", "title"));
+        map.forEach((stage, expectedFields) ->
+                assertEquals(expectedFields, MongoStatement.getFieldNamesFromProjectStage(stage)));
+    }
```

Comment:
Same number of lines, but less code duplication, and all checks are guaranteed to run, regardless if any fail.

```suggestion
    @Test
    void testGetProjectStageFieldNames() {
        BiConsumer<String, List<String>> asserter = (projectStage, expectedFieldNames) -> assertEquals(
                expectedFieldNames, MongoStatement.getFieldNamesFromProjectStage(BsonDocument.parse(projectStage)));
        assertAll(
                () -> asserter.accept("{title: 1, publishYear: 1}", List.of("title", "publishYear", "_id")),
                () -> asserter.accept("{title: 1, publishYear: 0}", List.of("title", "_id")),
                () -> asserter.accept("{title: 1, publishYear: false}", List.of("title", "_id")),
                () -> asserter.accept("{title: 1, _id: 0}", List.of("title")),
                () -> asserter.accept("{title: 1, _id: false}", List.of("title")),
                () -> asserter.accept("{_id: 1, title: 1}", List.of("_id", "title")));
    }
```